### PR TITLE
Validate athenz-domain in deployment.xml equal to tenant domain

### PR DESCRIPTION
**Pull request by tokle**

### CODE REVIEW

**Review by mpolden — APPROVED** [review]
LGTM

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java [hunk 1]
```diff
@@ -29,10 +31,12 @@
 
     private final ZoneRegistry zoneRegistry;
     private final AthenzClientFactory athenzClientFactory;
+    private final ApplicationPackage applicationPackage;
 
-    public DeployAuthorizer(ZoneRegistry zoneRegistry, AthenzClientFactory athenzClientFactory) {
+    public DeployAuthorizer(ZoneRegistry zoneRegistry, AthenzClientFactory athenzClientFactory, ApplicationPackage applicationPackage) {
```
Comment (bjorncs) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java]: Why is `ApplicationPackage` a constructor argument. I would make it an argument to `throwIfUnauthorizedForDeploy` instead.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java [hunk 2]
```diff
@@ -73,6 +77,19 @@ public void throwIfUnauthorizedForDeploy(Principal principal,
                         athenzPrincipal.toYRN(), applicationId, tenantDomain.id());
             }
         }
+
```
Comment (bjorncs) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/DeployAuthorizer.java]: Note the above return statement (line 47)!

**Review by bjorncs — COMMENTED** [review]
See comments

**Review by bjorncs — CHANGES_REQUESTED** [review]
See comments

**Comment by tokle** [comment]
@bjorncs ptal

**Review by bjorncs — APPROVED** [review]

**Pull request merged**